File: typescript/src/api/resources/workflows/client/requests/ImportWorkflowRequest.ts

/**
 * @example
 *     {
 *         payload: {
 *             "key": "value"
 *         }
 *     }
 */
export interface ImportWorkflowRequest {
    payload: Record<string, unknown>;
    sourceArtifactId?: string | null;
    workflowId?: string | null;
}

File: ruby/lib/Sikaru/workflows/types/import_workflow_request.rb
# frozen_string_literal: true

module Sikaru
  module Workflows
    module Types
      class ImportWorkflowRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :payload, -> { Internal::Types::Hash[String, Object] }, optional: false, nullable: false

        field :source_artifact_id, -> { String }, optional: true, nullable: false, api_name: "sourceArtifactId"

        field :workflow_id, -> { String }, optional: true, nullable: false, api_name: "workflowId"
      end
    end
  end
end
